tests/integration/utils.py

```diff
@@ -401,11 +401,13 @@ def build_image_and_check_cmd(
     assert exit_code == 0, f"Injecting project files failed. output-cmd: {output}"
 
     log.info("Build container image with all prerequisites retrieved in previous steps")
-    container_folder = test_data_dir.joinpath(test_case, "container")
 
-    with build_image_for_test_case(
-        tmp_path, str(container_folder.joinpath("Containerfile")), test_case
-    ) as test_image:
+    containerfile = tmp_path.joinpath("Containerfile")
+    if not containerfile.exists():
+        container_folder = test_data_dir.joinpath(test_case, "container")
+        containerfile = container_folder.joinpath("Containerfile")
+
+    with build_image_for_test_case(tmp_path, str(containerfile), test_case) as test_image:
```

Comment:
While I appreciate you're thinking of doing this in a compatible manner, it's actually going to lead to a very messy state where part of the tests will have the containerfile inside the e2e repo and the rest will have it in cachi2 integration test data. I'd prefer if this was proposed as a complex change == move the existing container files out of cachi2 and mandate that new e2e repos follow the proposed practice. Which brings me to #430 which this change would be nicely paired with that docs where this practice would be explained.